sycl/source/detail/thread_pool.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <algorithm>
+#include <atomic>
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <vector>
+
+#include <CL/sycl/detail/defines.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
```

Comment:
Isn't there a ThreadPool elsewhere in llvm that could be re-used?

Reply:
I'm not sure about it.

Reply:
https://llvm.org/doxygen/classllvm_1_1ThreadPool.html